src/act.drive.cpp

```diff
@@ -1815,27 +1816,87 @@ void process_autonav(void)
       if (!ch) {
         veh->dest = NULL;
         veh->cspeed = SPEED_OFF;
-        return;
+        continue;
       }
 
       int dir = 0;
-      for (int x = MIN(MAX((int)get_speed(veh) / 10, 1), MAX_GRIDGUIDE_ROOMS_PER_PULSE); x && dir >= 0 && veh->dest; x--) {
```

Comment:
I removed it because it made speed pointless due to how autonav works..

The initial problem is that there was no reason whatsover to own a fast vehicle. Speed was capped at 55 for all of them, which means that even the slowest of vehicles with an immense max load was moving at exactly the same speed, or EVEN faster than a fast vehicle which could crawl due to the load rule that was non-canon to begin with. But even removing the artificial 55 cap it still made almost no difference at all because of the max rooms per 3 seconds setting being that low.

The design intend is to remove the artificial cap and scale real speed according to vehicle speed. So that limiter had to go and a universal modifier applied vs vehicle speed directly. I performed tests (they're documented in the comment in config.h) but I was not aware that you can reach 400 speed (how? what mods? I thought you can only get like +20-30 speed via mods). This means the modifier needs to become heavier and/or perhaps apply an upper limit as well but closer to 25, not 10 because again we'd have all vehicles moving at the same speed or we'd have to make slow very vehicles extremely slow with a applying a heavier modifier.

The plan basically is to move autonav to worker thread so we can do this properly and apply a wait state *per room moved* depending on vehicle speed but did not go directly there because the vehicle/rigging system is getting overhauled and I can't plan ahead on mutex locks and thread safety before having the concrete design (or at least the basic skeleton).

Reply:
I want to add that "teleporting" can not be currently (as in as long as we run this every 3 seconds) avoided for very small distances and initiating autonav close to the next execution of process_autonav() but I don't think that it's inherently bad. You can *walk* 10 rooms in less time than it takes to go with a vehicle heh.

Reply:
Now that I think about it, an upper limit of 25 will just make those speed mods that can reach 400 speed irrelevant. Can you give me a configuration of mods that reaches 400 so I can make the fastest vehicle and test it with stock vehicles etc to find an acceptable modifier that doesn't make speed stuff useless until I can redesign this with wait states per room?

Reply:
You don't need mods to make a vehicle hit 400, as staff you can just edit the vehicle itself. It's what I do when I need test vehicles. I also don't have an issue with making this tick faster (every second? half-second?) and dividing the rooms traveled to match. Not sure what that would raise the effective speed minimum to though.

Reply:
Jack gave me a list of all the speed mod parts a few minutes ago so I'll build the fastest Saab possible and test with that.

Reply:
Oh one sidenote, regarding making this tick faster. That alone doesn't help us solve the issue because we can't apply wait states. Any wait states applied will block the main thread for x duration per room per vehicle moving on autonav lagging everything to hell and back. What it would do, without wait states, is it would just allow us to get more accurate measurements and predictable results when someone initiates autonav BUT at the "cost" of removing the up to 3 second delay per autopathing cycle, essentially making vehicles move WAY faster.

In fact depending on how my tests go, I may move process_autonav() to a slower interval, artificially increasing the minimum time required for autonav. That is if I can't settle on a speed modifier that works for the upper in-game possible limit and also doesn't make slow vehicles need 5 minutes to move. :P

Reply:
Or faster interval with even slower modifier resulting in less rooms per cycle.. Hmm I need to test to see what works but now that I have the means to get the highest possible speed in game, I can run proper tests and tune this.